async-36: New OctreeChunkLoader

pwinston opened the pull request:
# Description
* Create new `OctreeChunkLoader` class
    * In preparation for multi-level loader/renderering.
    * We want something outside of `OctreeImage` to manage multi-level loading/rendering.
    * `OctreeChunkLoader` is a starting point.
* Create new `LayerRef` tuple
    * So `OctreeChunkLoader` can submit chunk requests without a reference to OctreeLayer. 
    * So Layer depends on `OctreeChunkLoader` but not the reverse.
    * This is better anyway: so `ChunkLoader` depends less on layers.
        * We can probably break the dependency 100% with some more effort.
* A few random `pylint` fixes.

## Type of change
- [x] Refactoring in experimental code

pwinston committed:
Extract method: _get_drawable_chunks

pwinston committed:
Fix pylint warning about duplicate name//scope

pwinston committed:
Fix pylint, else not valid/needed here

pwinston committed:
Introduce OctreeChunkLoader and LayerRef

OctreeChunkLoader to factor out this logic from OctdreeImage in preparation of multi-level rendering.

LayerRef alllows OctreeChunkLoader to not depend on OctreeImage
Since all it needs is a LayerRef not the layer
Move chunk stuff from _info.py to _utils.py

pwinston committed:
Move OctreeChunkLoader to own file

pwinston committed:
Fix test for LayerKeys

pwinston committed:
Fix test and number key/layers to match

pwinston committed:
Docstrings, rename attribute

pwinston committed:
Fix test again

pwinston committed:
Comments

pwinston committed:
Move LayerKey to own file, create LayerKey.from_layer()

pwinston committed:
Better class comment

pwinston committed:
Doc string

pwinston committed:
Doc strings

pwinston committed:
Fix syntax in test

pwinston committed:
Docstr, type

pwinston committed:
Comments

sofroniewn reviewed (approved):
This looks good to me. It's nice to be passing around the whole layer less, and I think breaking that dependency eventually would be good.

I'm approving, but I will wait to merge until all tests pass. Looks like a couple of async specific ones are still failing.

pwinston committed:
Fix tests

pwinston committed:
Constrast "visible" and "drawable"

Visible means the chunk is in the view, while
drawable means the data is in memory and
can be drawn.

pwinston committed:
Trigger Build
